impala/hiveserver2.py

```diff
@@ -760,11 +851,42 @@ def fetch_results(service, operation_handle, hs2_protocol_version, schema=None,
                         value = Decimal(value)
                 row.append(value)
             rows.append(tuple(row))
-    else:
-        raise HiveServer2Error(
-            "Got HiveServer2 version {0}; expected V1 - V6".format(
-                TProtocolVersion._VALUES_TO_NAMES[hs2_protocol_version]))
-    return rows
+        return rows
+
+    def _is_columnar_protocol(self):
+        return (self.hs2_protocol_version ==
+                TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V6)
+
+    def _is_pre_columnar(self):
+        return self.hs2_protocol_version in _pre_columnar_protocols
+
+
+class ColumnBatch(object):
+
+    def __init__(self, columns):
+        self.columns = columns
+        self.length = len(columns[0]) if len(columns) > 0 else 0
+
+
+class Column(object):
+
+    def __init__(self, data_type, values, nulls):
+        self.data_type = data_type
+        self.values = values
+        self.nulls = nulls
+
+    def __len__(self):
+        return len(self.values)
+
+
+def fetch_results(service, operation_handle, hs2_protocol_version, schema=None,
+                  max_rows=1024, orientation=TFetchOrientation.FETCH_NEXT,
+                  columnar=False):
+    fetcher = HS2Fetch(service, operation_handle, hs2_protocol_version,
+                       schema=schema, max_rows=max_rows,
+                       columnar=columnar,
+                       orientation=orientation)
+    return fetcher.get_results()
```

Comment:
Maybe I've been looking at too much Java, but this fn will return two very different types of data depending on the `columnar` setting.  Is there a better way to unify these together?